eis_toolkit/exploratory_analyses/dbscan.py

```diff
@@ -0,0 +1,62 @@
+import geopandas as gdp
+import numpy as np
+from beartype import beartype
+from sklearn.cluster import DBSCAN
+
+from eis_toolkit.exceptions import EmptyDataFrameException, InvalidParameterValueException
+
+
+def _dbscan(data: gdp.GeoDataFrame, max_distance: float, min_samples: int) -> gdp.GeoDataFrame:
+
+    coordinates = list(data.geometry.apply(lambda geom: [geom.x, geom.y]))
+    dbscan = DBSCAN(eps=max_distance, min_samples=min_samples)
+
+    dbscan.fit(coordinates)
+    data["cluster"] = dbscan.labels_
+
+    core_indices = dbscan.core_sample_indices_
+    mask = np.zeros(len(coordinates), dtype=int)
+    mask[core_indices] = 1
+    data["core"] = mask
+
```

Comment:
Here I would rename the "mask" to something that describes the array better. Perhaps "core_data" or something